app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,63 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = current_user.recipes
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    @recipe = Recipe.find(params[:id])
```

Comment:
- [ ] you can solve N+1 problem by changing your recipe to something like this ` @recipe = Recipe.includes(:recipe_foods, :user).find(params[:id])`